test/Dialect/Arc/lower-verif-simulations.mlir

```diff
@@ -0,0 +1,83 @@
+// RUN: circt-opt --arc-lower-verif-simulations %s | FileCheck %s
+
+// CHECK: func.func private @exit(i32)
+
+// CHECK-LABEL: hw.module @verif.simulation.impl.Foo(
+// CHECK-SAME: in %clock : !seq.clock
+// CHECK-SAME: in %init : i1
+// CHECK-SAME: out done : i1
+// CHECK-SAME: out exit_code : i32
+verif.simulation @Foo {} {
+^bb0(%clock: !seq.clock, %init: i1):
+  // CHECK: [[TMP1:%.+]] = hw.constant true
+  // CHECK: [[TMP2:%.+]] = hw.constant 0 : i32
+  // CHECK: hw.output [[TMP1]], [[TMP2]] : i1, i32
+  %true = hw.constant true
+  %c0_i32 = hw.constant 0 : i32
+  verif.yield %true, %c0_i32 : i1, i32
+}
+
+// CHECK-LABEL: func.func @Foo()
+// CHECK: [[I0:%.+]] = hw.constant false
+// CHECK: [[I1:%.+]] = hw.constant true
+// CHECK: [[C0:%.+]] = seq.to_clock [[I0]]
+// CHECK: [[C1:%.+]] = seq.to_clock [[I1]]
+// CHECK: arc.sim.instantiate @verif.simulation.impl.Foo as [[A:%.+]] {
+// CHECK:   scf.execute_region {
+// CHECK:     arc.sim.set_input [[A]], "clock" = [[C0]]
+// CHECK:     arc.sim.set_input [[A]], "init" = [[I1]]
+// CHECK:     arc.sim.step [[A]]
+// CHECK:     arc.sim.set_input [[A]], "clock" = [[C1]]
+// CHECK:     arc.sim.step [[A]]
+// CHECK:     arc.sim.set_input [[A]], "clock" = [[C0]]
+// CHECK:     arc.sim.set_input [[A]], "init" = [[I0]]
+// CHECK:     arc.sim.step [[A]]
```

Comment:
I think the simulation op description didn't mention that done is not checked when init=1. Which, of course, makes sense, but might be worth mentioning.

Reply:
Good point. Added that.